### DISCUSSION TITLE
how to list all users based on their groups

### DISCUSSION THREAD

**Discussion by ahmaruff** [codeigniter4/shield#611] [Q&A] [answered]
I want query all users based on their groups. e.g list all users where `group = superadmin`. but when I try, it return errors `Unknown column 'groups' in 'where clause' `
```php
$userModel = new UserModel();
$users = $userModel->where('groups', 'superadmin')->findAll();
foreach($users as $u) {
       var_dump($u);
       echo "<hr>";
}
die();
```

so how to do this? thankyou

**Comment by kenjis** [comment]
Group data is stored in `auth_groups_users` table.
See https://github.com/codeigniter4/shield/blob/697f570277a4d2205852cacaae4837604b1c4b88/src/Database/Migrations/2020-12-28-223112_create_auth_tables.php#L112-L121

**Comment by lonnieezell** [comment, marked as answer]
Bonfire has implemented this as part of a filter, but here's the [relevant code](https://github.com/lonnieezell/Bonfire2/blob/develop/src/Users/Models/UserFilter.php#L64) which would translate to something like: 

```php
$users = $userModel
    ->select('users.*)
    ->join('auth_groups_users agu', 'agu.user_id = users.id')
    ->whereIn('agu.group', 'superadmin')
    ->findAll();
```

**Reply by ahmaruff** [reply]
thank you. this solution help me. i'all mark this as an answer :)
